app/setup.py

```diff
@@ -132,11 +140,15 @@ def create_app(  # noqa: C901  pylint: disable=too-complex, too-many-statements
     @application.before_request
     def before_request():  # pylint: disable=unused-variable
         request_id = str(uuid4())
-        logger.new(request_id=request_id)
+
+        if flask_request.method == "OPTIONS":
+            contextvars.clear_contextvars()
```

Comment:
What is the reasoning behind this?. Why clear for OPTIONS?

Reply:
Before every request we need to clear the contextvars bind. If we did not clear for the request `OPTIONS ` the contextvars wont bind.

Reply:
This was for some unit tests that were not staged and committed, reverted now.